templates/theories/linear_algebra.md

```diff
@@ -0,0 +1,106 @@
+# Maths in lean : linear algebra
+
+### semimodules, modules and vector spaces ###
+
+#### algebra.module ####
+
+This file defines the typeclass `semimodule R M`, which gives an `R`-semimodule structure on the type `M`, and similarly `module R M` and `vector_space R M`.
+An additive commutative monoid `M` is a semimodule over the semiring `R` if there is a scalar multiplication `•` (`has_scalar.smul`) that satisfies the expected distributivity axioms for `+` (in `M` and `R`) and `*` (in `R`).
+A module (typeclass `module`) is a semimodule that additionally requires that `R` is a ring and `M` is a group.
+A vector space (typeclass `vector_space`) is a module that additionally requires that `R` is a field.
+All vector spaces are also modules, and all modules are also semimodules.
+
+Typical examples are that `m → ℕ` is an `ℕ`-semimodule, `m → ℤ` is a `ℤ`-module and `m → ℚ` is a `ℚ`-vector space
+(outside of type theory, these are known as `ℕ^m`, `ℤ^m` and `ℚ^m` respectively).
+These instances are defined in `algebra.pi_instances`.
```

Comment:
I think all those references to file names should be links to the docs. For instance [algebra.pi_instances](https://leanprover-community.github.io/mathlib_docs/algebra/pi_instances.html)